extensions/funding/service.go

```diff
@@ -300,63 +188,24 @@ func (s service) DeriveFromUpdatePayload(ctx context.Context, req *clientfunpb.F
 	return model, nil
 }
 
+// TODO: Move to attribute utils
 func (s service) findFunding(model documents.Model, fundingID string) (*Data, error) {
-	idx, err := s.findFundingIDX(model, fundingID)
+	idx, err := extensions.FindAttributeSetIDX(model, fundingID, fundingLabel, agreementIDLabel, fundingFieldKey)
 	if err != nil {
 		return nil, err
 	}
 	return s.deriveFundingData(model, idx)
 }
 
-func (s service) findFundingIDX(model documents.Model, agreementID string) (idx string, err error) {
-	lastIdx, err := getArrayLatestIDX(model, fundingLabel)
-	if err != nil {
-		return idx, err
-	}
-
-	i, err := documents.NewInt256("0")
-	if err != nil {
-		return idx, err
-	}
-
-	for i.Cmp(lastIdx) != 1 {
-		label := generateLabel(fundingFieldKey, i.String(), agreementIDLabel)
-		k, err := documents.AttrKeyFromLabel(label)
-		if err != nil {
-			return idx, err
-		}
-
-		attr, err := model.GetAttribute(k)
-		if err != nil {
-			return idx, err
-		}
-
-		attrFundingID, err := attr.Value.String()
-		if err != nil {
-			return idx, err
-		}
-		if attrFundingID == agreementID {
-			return i.String(), nil
-		}
-		i, err = i.Inc()
-
-		if err != nil {
-			return idx, err
-		}
-
-	}
-
-	return idx, ErrFundingNotFound
-}
-
+// TODO: Move to attribute utils
 func (s service) deriveFundingData(model documents.Model, idx string) (*Data, error) {
 	data := new(Data)
 
```

Comment:
I think we can move the entire deriveFundingData to the extension package. With a data parameter `interface{}`

After that the `extensions.LabelFromJSONTag` can be private instead of public

Reply:
It is not a requirement for this PR